docs/csharp/whats-new/breaking-changes.md

```diff
@@ -1,13 +1,189 @@
+
+***Introduced in .NET SDK 6.0.200, Visual Studio 2022 version 17.1.***
+
+The `Length` and `Count` properties, even though typed as `int`, are assumed to be non-negative when analyzing patterns. Consider this sample method:
+
+```csharp
+string SampleSizeMessage<T>(IList<T> samples)
+{
+    return samples switch
+    {
+        { Count: < 0 }    => throw new InvalidOperationException(), // Can't happen!
+        { Count:  0 }     => "Empty collection",
+        { Count: < 5 }    => "Too small",
+        { Count: < 20 }   => "reasonable for the first pass",
+        { Count: < 100 }  => "reasonable",
+        { Count: >= 100 } => "fine",
+    };
+}
+```
+
+Prior to 17.1, The first switch arm, testing that `Count` is negative was necessary to avoid a warning that all possible values weren't covered. Starting with 17.1, the first switch arm generates a compiler error. The workaround is to remove the switch arms added for the invalid cases.
+
+This change was added as part of adding list patterns. The processing rules are more consistent if every use of a `Length` or `Count` property on a collection are considered non-negative. You can read more details about the change in the [language design issue](https://github.com/dotnet/csharplang/issues/5226).
+
+The workaround is to remove the switch arms with unreachable conditions.
+
+## Adding field initializers to a struct requires an explicitly declared constructor
+
+***Introduced in .NET SDK 6.0.200, Visual Studio 2022 version 17.1.***
+
+There are two ways to initialize a variable to its default value in C#: `new()` and `default`. For classes, the difference is evident since `new` creates a new instance and `default` returns `null`. The difference is more subtle for structs, since for `default`, structs return an instance with each field/property set to its own default. We added field initializers for structs in C# 10. Field initializers are executed only when an explicitly declared constructor runs. Significantly, they don't execute when you use `default` or create an array of any `struct` type.
+
+In earlier versions, if there are field initializers but no declared constructors, a parameterless constructor is synthesized that runs field initializers. However, that meant adding or removing a constructor declaration may affect whether a parameterless constructor is synthesized, and as a result, may change the behavior of `new()`.
+
+To address the issue, in .NET SDK 6.0.200 (VS 17.1) the compiler no longer synthesizes a parameterless constructor. If a `struct` contains field initializers and no explicit constructors, the compiler generates an error. If a `struct` has field initializers it must declare a constructor, because otherwise the field initializers are never executed.
+
+The workaround is to declare a constructor. This constructor can, and often will, be an empty parameterless constructor:
+
+```c#
+struct S
+{
+    public S() { }
+    // remainder of struct code
+}
+```
```

Comment:
Consider expanding slightly and including a before and after. Perhaps:

To address the issue, in .NET SDK 6.0.200 (VS 17.1) the compiler no longer synthesizes a parameterless constructor. If a `struct` contains field initializers and no explicit constructors, the compiler generates an error. If a `struct` has field initializers it must declare a constructor, because otherwise the field initializers are never executed.

Additionally, all fields must be definitely assigned in `struct` constructors that do not have a `: this()` initializer, so any previously unassigned fields must assigned from the constructor or from field initializers.

For instance:
```c#
struct S // error CS8983: A 'struct' with field initializers must include an explicitly declared constructor.
{
    int X = 1; 
    int Y;
}
```

The workaround is to declare a constructor. Unless fields were previously unassigned, this constructor can, and often will, be an empty parameterless constructor:
```c#
struct S
{
    int X = 1;
    public S() { Y = 0; } // ok
}
```